WalletWasabi.Fluent/ViewModels/Settings/GeneralSettingsTabViewModel.cs

```diff
@@ -96,5 +96,11 @@ public GeneralSettingsTabViewModel()
 		protected override void EditConfigOnSave(Config config)
 		{
 		}
+
+		protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+		{
+			_runOnSystemStartup = Services.UiConfig.RunOnSystemStartup;
```

Comment:
Should be `RunOnSystemStartup` instead of `_runOnSystemStartup`.

Setting _runOnSystemStartup has no effect in UI because it is the backing filed. RaisePropertyChanged never gets called.
While setting RunOnSystemStartup will call RaisePropertyChanged.

Set the backing field in ctor because in reference type cases it eliminates null warning. In every other place, the property should be set.

Although you won't see any difference because currently it is bugged. `OnNavigatedTo` never gets called. It is because we are actually not navigating to those pages. I will open an issue for it.

Reply:
You are right, `OnNavigatedTo` never gets called.
Sadly, setting the backing field in ctor of the `ViewModel` won't help me here, bc the ctor runs before the `InitializeNoWalletAsync` method (`Global.cs`) (that's where the "checking" gets called currently.)

Possible solutions could be:

- Call my checker function from the ctor of the `Global.cs`
- Call my checker function when we load the Config Files (`LoadOrCreateConfigs`)

Reply:
Sorry I was not clear. I meant it in general... set the backing field if you are in the ctor and set the prop if you are everywhere else.

Anyways... I didn't follow this PR but if you take a look at the `UiConfig`, and the `WindowState` in it. A converter like `WindowStateAfterStartJsonConverter` could help you? I am not sure but worth a look

Reply:
Thank you for the clarification!
Unfortunately `WindowState` won't help me here.
I will come up with a different solution.

Reply:
What I meant is,

create a converter like this:
```
	public class RunOnSystemStartupAfterStartJsonConverter : JsonConverter
	{
		public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
		{
			writer.WriteValue(value);
		}

		public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
		{
			using var cts = new CancellationTokenSource(10000);
			var value = StartupChecker.GetCurrentValueAsync(cts.Token).Result;

			return value;
		}

		public override bool CanConvert(Type objectType)
		{
			return objectType == typeof(bool);
		}
	}
```

And use it in `UiConfig` like
```
		[DefaultValue(false)]
		[JsonProperty(PropertyName = "RunOnSystemStartup", DefaultValueHandling = DefaultValueHandling.Populate)]
		[JsonConverter(typeof(RunOnSystemStartupAfterStartJsonConverter))]
		public bool RunOnSystemStartup
		{
			get => _runOnSystemStartup;
			set => RaiseAndSetIfChanged(ref _runOnSystemStartup, value);
		}
```

This will set `RunOnSystemStartup` value to `StartupChecker.GetCurrentValueAsync(cts.Token).Result`'s value when the UiConfig gets created. But again, I am not familiar with the exact problem, this is might not useful, if so just leave it. At first glance, it looks like a possible solution.

Reply:
Will check out. Thanks!

Reply:
This was exactly what i needed. Thank you!

Reply:
You're welcome.

BTW do we really need to serialize `RunOnSystemStartup`? Seemingly it makes no sense because every time when UiConfig is created basically its value just comes from `StartupChecker.GetCurrentValueAsync(cts.Token).Result`, nobody cares about the value what is in the json file.